xyz-psql-connector/src/main/java/com/here/xyz/psql/DatabaseHandler.java

```diff
@@ -165,50 +171,51 @@ void reset() {
     @Override
     protected synchronized void initialize(Event event) {
         this.event = event;
-        String ecps = PSQLConfig.getECPS(event);
+        this.config= new PSQLConfig(event, context, streamId);
+        final String dbStringConfig = config.getConfigValuesAsString();
 
-        if( PSQLConfig.DEFAULT_ECPS.equals(ecps) && context instanceof SimulatedContext )
-         ecps = idFromPsqlEnv((SimulatedContext) context);
+        boolean resourceInitFinished;
+        resourceInitFinished = dbInstanceMap.containsKey(dbStringConfig);
 
-        if (!dbInstanceMap.containsKey(ecps)) {
-            /** Init dataSource, readDataSource ..*/
-            logger.debug("{} - Create new config and data source for ECPS string: '{}'", streamId, ecps);
-            final PSQLConfig config = new PSQLConfig(event, context);
-            final String sName   = ecps.length() <  8 ? ecps : ( ecps.length() < 33 ? ecps.substring(0, 7) : ecps.substring(21, 28) ),
-                         appName = String.format("%s[%s]", config.applicationName(), sName );
+        if(resourceInitFinished){
+            /** Check if db-params has changed*/
+            if(!dbInstanceMap.get(dbStringConfig).getDatabaseConfig().equalsIgnoreCase(config.getConfigValuesAsString())) {
```

Comment:
Please correct me if I'm wrong. This method being synchronized and the only one writing into dbInstanceMap (despite of reset() is not in synch yet with initialize()), wouldn't be impossible to enter this if statement? Since the string resulting from getDatabaseConfig() is the same string generated by getConfigValuesAsString, therefore the entire block from line 180 until 186 is unnecessary.